### PR TITLE
feat: not start check in local

### PR DESCRIPTION
**Pull request by whxaxes**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

https://github.com/eggjs/egg/issues/2633

我觉得可以直接在 local 下，就不启用检查，所以就按照这么改，看一下合不合适？

### CODE REVIEW

File: lib/master.js [hunk 1]
```diff
@@ -90,7 +90,9 @@ class Master extends EventEmitter {
       this.messenger.send({ action, to: 'agent', data: this.options });
 
       // start check agent and worker status
-      this.workerManager.startCheck();
+      if (process.env.EGG_SERVER_ENV !== 'local') {
```
Comment (dead-horse) [inline lib/master.js]: `if (!this.isProduction)`

Comment (dead-horse) [reply lib/master.js]: others LGTM

Comment (whxaxes) [reply lib/master.js]: Only start check in production ? It seems like an breakchange.

File: test/master.test.js [hunk 2]
```diff
@@ -654,14 +654,26 @@ describe('test/master.test.js', () => {
...
+      yield sleep(30000);
+
+      // process should exist
+      assert(app.process.exitCode === null);
+      app.process.kill('SIGINT');
+    });
+
+    it('should exit when no agent after check 3 times', function* () {
+      app = utils.cluster('apps/check-status');
+      app.debug();
```
Comment (atian25) [inline test/master.test.js]: comment this

Comment (whxaxes) [reply test/master.test.js]: done

File: lib/master.js (hunk 1 above)
Comment (dead-horse) [reply lib/master.js]: https://github.com/eggjs/egg-cluster/blob/3621081eeb7eba35f1ee97bb56c0a4d7d829e24f/lib/master.js#L537

File: test/fixtures/apps/check-status/app.js [hunk 3]
```diff
@@ -1,5 +1,13 @@
 'use strict';
 
-process.on('message', function(msg) {
-  if (msg.action === 'kill') process.exit(1);
-});
+const fs = require('fs');
+const path = require('path');
+
+module.exports = app => {
+  if (fs.existsSync(path.join(app.baseDir, 'logs/started'))) {
```
Comment (atian25) [inline test/fixtures/apps/check-status/app.js]: why not notify app by message like below? why do use file ?

Comment (whxaxes) [reply test/fixtures/apps/check-status/app.js]: Should exit after three times checking.

File: lib/master.js (hunk 1 above)
Comment (whxaxes) [reply lib/master.js]: I found you said change to `if (!this.isProduction)` , It seems change to `if (this.isProduction)` more reasonable? Only start check while `this.isProduction` was true, Am I right?

**Comment by whxaxes** [comment]
Any progress？

**Comment by dead-horse** [comment]
1.17.0

**Pull request merged**